wip: many_partitions_test with cloud topics

andrwng opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* None
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

andrwng committed:
many_partitions_test: expand wait time for leadership balancing

I saw one instance where, run locally, the test would wait 62 seconds
for leadership to quiesce. This doesn't didn't give much buffer room at
all for actual balancing, since the balancing tick may happen once
every 30 seconds and nodes are considered "muted" for 30 seconds if
they have been down, as is the case for this test.

This commit bumps the time a bit with another 30 second interval.

andrwng committed:
ct/l0: use STM prefix logger in ctp_stm_api

andrwng committed:
many_partitions_test: test case for cloud topics

Adds a test case of many_partitions_test with cloud topics. This doesn't
run through the warmup phase like tiered storage -- instead, it's closer
to the non-tiered-storage test, but with slightly relaxed performance
expectations.

andrwng committed:
many_partitions_test: bump logging

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--b2f42b57-4945-4d67-a001-9e72e7612aff-->
<!--TEST RESULTS-->
<!--f6a99207-174d-44d9-b10f-0a54ca0917ff-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75099">build#75099</a></summary>


| test_class                | test_method                          | test_arguments        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   | test_history                                                                                                                                                             |
|:--------------------------|:-------------------------------------|:----------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest | test_write                           | null                  | integration | https://buildkite.com/redpanda/redpanda/builds/75099#019a2a18-0348-4f1d-905c-d6993884d673 | FLAKY         | 19/21    | upstream reliability is '96.10619469026548'. current run reliability is '90.47619047619048'. drift is 5.63 and the allowed drift is set to 50. The test should PASS      | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                       |
| DataMigrationsApiTest     | test_creating_and_listing_migrations | null                  | integration | https://buildkite.com/redpanda/redpanda/builds/75099#019a2a11-1614-4bcd-9e1a-19f5b4a4c483 | FLAKY         | 13/21    | upstream reliability is '99.69465648854961'. current run reliability is '61.904761904761905'. drift is 37.78989 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
| PartitionMovementTest     | test_empty                           | {"num_to_upgrade": 0} | integration | https://buildkite.com/redpanda/redpanda/builds/75099#019a2a11-161c-4e46-ab91-f7dd9949b4a4 | FLAKY         | 17/21    | upstream reliability is '90.22988505747126'. current run reliability is '80.95238095238095'. drift is 9.2775 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                           |
| PartitionMovementTest     | test_empty                           | {"num_to_upgrade": 0} | integration | https://buildkite.com/redpanda/redpanda/builds/75099#019a2a18-034c-4111-9a2e-e733b303c21b | FLAKY         | 19/21    | upstream reliability is '90.22988505747126'. current run reliability is '90.47619047619048'. drift is -0.24631 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                           |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75145">build#75145</a></summary>


| test_class         | test_method                       | test_arguments                                                       | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 | test_history                                                                                                                                                       |
|:-------------------|:----------------------------------|:---------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ManyPartitionsTest | test_many_partitions_cloud_topics | {"mib_per_partition": 0.1953125, "topic_partitions_per_shard": 3000} | integration | https://buildkite.com/redpanda/redpanda/builds/75145#019a2c28-e5bf-42b5-a37d-6ed141b6631f | FAIL          | 0/30     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ManyPartitionsTest&test_method=test_many_partitions_cloud_topics |
| ManyPartitionsTest | test_many_partitions_cloud_topics | {"mib_per_partition": 0.1953125, "topic_partitions_per_shard": 3000} | integration | https://buildkite.com/redpanda/redpanda/builds/75145#019a2c28-e5bf-477e-ac81-df23328d1255 | FAIL          | 0/30     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ManyPartitionsTest&test_method=test_many_partitions_cloud_topics |
| ManyPartitionsTest | test_many_partitions_cloud_topics | {"mib_per_partition": 0.1953125, "topic_partitions_per_shard": 3000} | integration | https://buildkite.com/redpanda/redpanda/builds/75145#019a2c28-e5bf-4bb6-97b3-6cb3837fa8ea | FAIL          | 0/30     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ManyPartitionsTest&test_method=test_many_partitions_cloud_topics |
</details>
<!--/body-->

andrwng commented:
/ci-repeat 3
dt-repeat=10
skip-redpanda-build
skip-units
tests/rptest/tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng commented:
/ci-repeat 3
dt-repeat=10
skip-redpanda-build
skip-units
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75145-->
<!--39f24662-2c65-4fea-81ad-1a195d078116-->
<!--3b47bbdf-d5df-48b2-af02-8cf335f1249f-->
<!--05ed6816-085c-4c21-9257-dcbf175cf62f-->
<!--title-->
## Retry command for [Build#75145](https://buildkite.com/redpanda/redpanda/builds/75145)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics@{"mib_per_partition":0.1953125,"topic_partitions_per_shard":3000}
```
<!--/body-->

andrwng commented:
/cdt
dt-test-timeout=10800000
dt-repeat=2
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng commented:
/cdt
dt-test-timeout=10800000
dt-repeat=2
rp_version=build
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng commented:
/cdt
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng committed:
ct: make reconciliation interval tunable property

andrwng committed:
ct: make L1 GC interval configurable

andrwng commented:
/cdt
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng committed:
ct: minor logging improvements

- I found it helpful to be able to tell if the housekeeper was doing any
  work (in particular, I incorrectly thought we were incorrectly
  deleting objects, and understanding if the housekeeper was making
  progress with deletion would have been helpful)
- I found it helpful to understand what state updates are being appplied
  in L1
- I also found it helpful to be able to tell if the reconiliation loop
  has exited (I was debugging a shutdown hang)

andrwng committed:
ct/l1: don't replicate on no-op set_start_offset

andrwng committed:
ct: demote some error logs to warning

May happen at shutdown, or due to transient error.

andrwng committed:
ct: exception

andrwng committed:
mpt: frequent GC

andrwng committed:
mpt: logging

andrwng commented:
/cdt
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng committed:
ct/l1: rename frontend to leader_router

andrwng committed:
ct/l1: rename frontend class to leader_router

andrwng committed:
ct/l1: add logging to leader router

Adds some logging to help debug issues that may happen with in-flight
requests. This was helpful in identifying the root cause of a shutdown
hang, where the hang happened in the middle of an add_objects call.

andrwng committed:
ct: move shutdown to after partition shutdown

In the many partitions test with cloud topics, Redpanda would
occasionally hang when shutting down. This was because we would wait for
a Raft replication attempt to finish when shutting down the L1 leader
router, but it wouldn't complete with the RPC subsystem shutdown already
happening.

The solution is to shut down the partitions first, which in turn stops
further Raft replication. This is the suggested solution for other STMs,
as indicated by nearby comments in application.cc.

andrwng commented:
/cdt
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

andrwng committed:
kgo_repeater_services: bump start timeout

I saw it take 7 seconds to start up with CDT. Bumps it to a conservative
30 seconds.

andrwng commented:
/cdt
tests/rptest/scale_tests/many_partitions_test.py::ManyPartitionsTest.test_many_partitions_cloud_topics

micheleRP reviewed (approved):
looks good from docs
